src/spatialdata/_core/query/spatial_query.py

```diff
@@ -102,6 +104,165 @@ def _get_bounding_box_corners_in_intrinsic_coordinates(
+
+    Returns
+    -------
+    m_without_c
+        The transformation from the element's intrinsic coordinate system to the query coordinate space, without the
+        "c" axis.
+    input_axes_without_c
+        The axes of the element's intrinsic coordinate system, without the "c" axis.
+    output_axes_without_c
+        The axes of the query coordinate system, without the "c" axis.
+
+    """
+    from spatialdata.transformations import get_transformation
+
+    transform_to_query_space = get_transformation(element, to_coordinate_system=target_coordinate_system)
+    assert isinstance(transform_to_query_space, BaseTransformation)
+    m = _get_affine_for_element(element, transform_to_query_space)
+    input_axes_without_c = tuple([ax for ax in m.input_axes if ax != "c"])
+    output_axes_without_c = tuple([ax for ax in m.output_axes if ax != "c"])
+    m_without_c = m.to_affine_matrix(input_axes=input_axes_without_c, output_axes=output_axes_without_c)
+    return m_without_c, input_axes_without_c, output_axes_without_c
+
+
+def _adjust_bounding_box_to_real_axes(
+    axes: tuple[str, ...],
+    min_coordinate: ArrayLike,
+    max_coordinate: ArrayLike,
+    output_axes_without_c: tuple[str, ...],
+) -> tuple[tuple[str, ...], ArrayLike, ArrayLike]:
+    """
+    Adjust the bounding box to the real axes of the transformation.
+
+    The bounding box is defined by the user and it's axes may not coincide with the axes of the transformation.
+    """
+    if set(axes) != set(output_axes_without_c):
+        axes_only_in_bb = set(axes) - set(output_axes_without_c)
+        axes_only_in_output = set(output_axes_without_c) - set(axes)
+
+        # let's remove from the bounding box whose axes that are not in the output axes (e.g. querying 2D points with a
+        # 3D bounding box)
+        indices_to_remove_from_bb = [axes.index(ax) for ax in axes_only_in_bb]
+        axes = tuple([ax for ax in axes if ax not in axes_only_in_bb])
+        min_coordinate = np.delete(min_coordinate, indices_to_remove_from_bb)
+        max_coordinate = np.delete(max_coordinate, indices_to_remove_from_bb)
+
+        # if there are axes in the output axes that are not in the bounding box, we need to add them to the bounding box
+        # with a range that includes everything (e.g. querying 3D points with a 2D bounding box)
+        for ax in axes_only_in_output:
+            axes = axes + (ax,)
+            M = np.finfo(np.float32).max - 1
+            min_coordinate = np.append(min_coordinate, -M)
+            max_coordinate = np.append(max_coordinate, M)
+    return axes, min_coordinate, max_coordinate
+
+
+def _get_case_of_bounding_box_query(
+    m_without_c_linear: ArrayLike,
+    input_axes_without_c: tuple[str, ...],
+    output_axes_without_c: tuple[str, ...],
+) -> int:
+    """Get the case of the current bounding box query.
+
+    See https://github.com/scverse/spatialdata/pull/151 for a detailed overview of the logic of this code,
```

Comment:
I think that description was really cool and maybe it's worth to add it as a living document somewhere evnetually

Reply:
Please link to the specific comment you are referring to. That way people know where to look.

This is a private function, so I am okay with leaving it  less documented. However, I think this would be tough for a maintainer in the future to understand what's going on. Can you please add some comments throughout to explain what is going on?

Reply:
I lnked the specific comment, also we can use GitHub as a living document as the core devs can edit the comments of other people.